### PR TITLE
Fix meta_yml linting error

### PR DESCRIPTION
**Pull request by lmReef**
## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated

## Description

This PR is to fix an issue where running `nf-core modules lint -a -k meta_yml` always has failing tests due to `module.process_name` always being `""` in [nfcore_component.py](https://github.com/nf-core/tools/blob/main/nf_core/components/nfcore_component.py).

It gets initialized to `""` then eventually gets updated in `main_nf` but not in any others which results in the failed `meta_yml` tests unless `-k main_nf` is specified as well

I've added a simple function for the component to set it's `process_name` itself rather than it being set in other separate parts of the code. It uses the same regex expression that `main_nf.py` uses

### CODE REVIEW

**Comment by lmReef** [comment]
I think the issue was added [here](https://github.com/nf-core/tools/commit/200ae85466eef6f544227bf9556be686c2161946) when the test was added back, but it was not so obvious that `module.process_name` can sometimes still be unset in that context

**Review by mirpedrol — APPROVED** [review]
LGTM, tests are currently failing due to a Java version, a fix is on the way, we can merge once that is fixed 👍 
Thanks for this PR!

File: nf_core/components/nfcore_component.py [hunk 1]
```diff
@@ -96,6 +95,9 @@ def __init__(
             self.test_yml = None
             self.test_main_nf = None
 
+        # Set process_name after self.main_nf is defined
+        self.process_name = self._get_process_name()
```
[the remark this answers is not in the document]
Comment (lmReef) [reply nf_core/components/nfcore_component.py]: Thanks I missed that! added.

[the remark this answers is not in the document]
Comment (lmReef) [reply nf_core/components/nfcore_component.py]: @mirpedrol is this good to go now with https://github.com/nf-core/tools/commit/ce866c62c141ce1e0126dad9942f879e8bea3f89?

**Pull request merged**